x-pack/plugins/upgrade_assistant/server/lib/reindexing/index_settings.ts

```diff
@@ -83,23 +79,60 @@ export const getReindexWarnings = (flatSettings: FlatSettings): ReindexWarning[]
 
 const removeUnsettableSettings = (settings: FlatSettings['settings']) =>
   omit(settings, [
-    'index.uuid',
+    // Private ES settings
+    'index.allocation.existing_shards_allocator',
     'index.blocks.write',
     'index.creation_date',
-    'index.legacy',
-    'index.mapping.single_type',
+    'index.frozen',
+    'index.history.uuid',
+    'index.merge.enabled',
     'index.provided_name',
+    'index.resize.source.name',
+    'index.resize.source.uuid',
     'index.routing.allocation.initial_recovery._id',
+    'index.search.throttled',
+    'index.source_only',
+    'index.shrink.source.name',
+    'index.shrink.source.uuid',
+    'index.store.snapshot.repository_name',
+    'index.store.snapshot.snapshot_name',
+    'index.store.snapshot.snapshot_uuid',
+    'index.store.snapshot.index_name',
+    'index.store.snapshot.index_uuid',
+    'index.uuid',
+    'index.verified_before_close',
     'index.version.created',
+
+    // Deprecated in 9.0
     'index.version.upgraded',
-    'index.verified_before_close',
   ]);
 
+const validateSettings = (settings: FlatSettings['settings']) => {
```

Comment:
This follows the [same pattern used for the 6-7 upgrade for deprecated settings](https://github.com/elastic/kibana/blob/6.8/x-pack/plugins/upgrade_assistant/server/lib/reindexing/index_settings.ts#L158). In a follow-up PR, I think it's worth rethinking this. I think it might be a better experience if we add a warning to a user indicating they are using a deprecated setting, and then automatically delete it for them if they choose to proceed.